tests: Could not find token

https://ci.deis.io/job/test-ec2/128/console

```
/var/lib/jenkins/jobs/test-ec2/workspace/src/github.com/deis/deis/tests
Could not find token. Use `deis login` or `deis register` to get started.
error at command wait
--- FAIL: TestBuilds (86.49s)
	builds_test.go:64: exit status 1
```

--- a/tests/bin/test-integration-ec2.sh
+++ b/tests/bin/test-integration-ec2.sh
@@ -68,7 +68,7 @@ fi
 
 make discovery-url
 # add random characters after STACK_TAG to avoid collisions
-STACK_TAG=${STACK_TAG:-test}-$(openssl rand -hex 4)
+STACK_TAG=${STACK_TAG:-test}-$DEIS_TEST_ID
 STACK_NAME=deis-$STACK_TAG
 echo "Creating CloudFormation stack $STACK_NAME"
 $DEIS_ROOT/contrib/ec2/provision-ec2-cluster.sh $STACK_NAME

--- a/tests/bin/test-setup.sh
+++ b/tests/bin/test-setup.sh
@@ -50,6 +50,14 @@ echo "HOST_IPADDR=$HOST_IPADDR"
 export DEV_REGISTRY=${DEV_REGISTRY?}
 echo "DEV_REGISTRY=$DEV_REGISTRY"
 
+# random 10-char (5-byte) hex string to identify a test run
+export DEIS_TEST_ID=${DEIS_TEST_ID:-$(openssl rand -hex 5)}
+echo "DEIS_TEST_ID=$DEIS_TEST_ID"
+
+# give this session a unique ~/.deis/<client>.json file
+export DEIS_PROFILE=test-$DEIS_TEST_ID
+rm -f $HOME/.deis/test-$DEIS_TEST_ID.json
+
 # bail if registry is not accessible
 if ! curl -s $DEV_REGISTRY; then
   echo "DEV_REGISTRY is not accessible, exiting..."
@@ -76,9 +84,6 @@ test -e ~/.ssh/deiskey || ssh-keygen -q -t rsa -f ~/.ssh/deiskey -N '' -C deiske
 ssh-add -D || eval $(ssh-agent) && ssh-add -D
 ssh-add ~/.ssh/$DEIS_TEST_AUTH_KEY
 ssh-add $DEIS_TEST_SSH_KEY
-
-# clean out deis session data
-rm -rf ~/.deis
 
 # wipe out all vagrants & deis virtualboxen
 function cleanup {

--- a/tests/integration_test.go
+++ b/tests/integration_test.go
@@ -5,14 +5,13 @@ package tests
 import (
 	"encoding/json"
 	"io/ioutil"
+	"os"
 	"os/user"
 	"path"
 	"testing"
 
 	"github.com/deis/deis/tests/utils"
 )
-
-const clientJsonFilePath string = ".deis/client.json"
 
 var (
 	gitCloneCmd  = "if [ ! -d {{.ExampleApp}} ] ; then git clone https://github.com/deis/{{.ExampleApp}}.git ; fi"
@@ -39,6 +38,11 @@ func clientTest(t *testing.T, params *utils.DeisTestConfig) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	profile := os.Getenv("DEIS_PROFILE")
+	if profile == "" {
+		profile = "client"
+	}
+	clientJsonFilePath := ".deis/" + profile + ".json"
 	data, err := ioutil.ReadFile(path.Join(user.HomeDir, clientJsonFilePath))
 	if err != nil {
 		t.Fatal(err)

